conans/client/generators/msbuild.py

```diff
@@ -0,0 +1,208 @@
+        else:
+            content_multi = self.dep_props
+
+        # parse the multi_file and add a new import statement if needed
+        dom = minidom.parseString(content_multi)
+        import_group = dom.getElementsByTagName('ImportGroup')[0]
+        children = import_group.getElementsByTagName("Import")
+        for node in children:
+            if (name_conf == node.getAttribute("Project") and
+                    condition == node.getAttribute("Condition")):
+                # the import statement already exists
+                break
+        else:
+            # create a new import statement
+            import_node = dom.createElement('Import')
+            import_node.setAttribute('Condition', condition)
+            import_node.setAttribute('Project', name_conf)
+            # add it to the import group
+            import_group.appendChild(import_node)
+        content_multi = dom.toprettyxml()
+        content_multi = "\n".join(line for line in content_multi.splitlines() if line.strip())
+        return content_multi
+
+    def _general(self, name_general, deps, conf_name, condition):
+        # read the existing mult_filename or use the template if it doesn't exist
+        multi_path = os.path.join(self.output_path, name_general)
+        if os.path.isfile(multi_path):
+            content_multi = load(multi_path)
+        else:
+            content_multi = self.dep_props
+
+        # parse the multi_file and add a new import statement if needed
+        dom = minidom.parseString(content_multi)
+        import_group = dom.getElementsByTagName('ImportGroup')[0]
+        children = import_group.getElementsByTagName("Import")
+        for dep in deps:
+            conf_props_name = "conan_%s%s.props" % (dep, conf_name)
+            for node in children:
+                if (conf_props_name == node.getAttribute("Project") and
+                        condition == node.getAttribute("Condition")):
+                    # the import statement already exists
+                    break
+            else:
+                # create a new import statement
+                import_node = dom.createElement('Import')
+                import_node.setAttribute('Condition', condition)
+                import_node.setAttribute('Project', conf_props_name)
+                # add it to the import group
+                import_group.appendChild(import_node)
+        content_multi = dom.toprettyxml()
+        content_multi = "\n".join(line for line in content_multi.splitlines() if line.strip())
+        return content_multi
+
+    def _dep_conf(self, name, cpp_info, conf_name):
+        def has_valid_ext(lib):
+            ext = os.path.splitext(lib)[1]
+            return ext in VALID_LIB_EXTENSIONS
+
+        t = "<Import Project=\"{}\" Condition=\"'$(conan_{}_props_imported)' != 'True'\"/>"
+        transitive_imports = []
+        for dep_name in cpp_info.public_deps:
+            conf_props_name = "conan_%s%s.props" % (dep_name, conf_name)
```

Comment:
Maybe it easier to add here the `"conan_%s.props" % dep_name` instead of the one matching the configuration. The logic inside that file will do the rest.

Reply:
I think it would be a bit confusing, as it will always resolve to one and only one of those conditionals, so better remove the intermediate steps. I like a xxxx_release_v141.props to import only other yyyy_release_v141.props, and not just yyyy.props, that could in theory resolve to yyyy_debug_v151.props if the conditionals inside are changed or something.

Reply:
This is related to the comment in the PR (https://github.com/conan-io/conan/pull/7035#pullrequestreview-418233801): define only the Conan variables in the `conan_zlib_xxxxxx_yyyyy.props` file, import that file in `conan_zlib.props` and populate the  VS ones here